react/utils/pages.ts

```diff
@@ -325,18 +325,29 @@ export function navigate(
     fetchPage = true,
     preventRemount,
     skipSetPath = false,
+    setPreview,
   } = options
 
   const navigationRoute = getNavigationRouteToNavigate(pages, options, true)
+
   if (navigationRoute.hash) {
     window.location.hash = navigationRoute.hash
+
     return true
   }
 
   if (navigationRoute == null) {
     return false
   }
 
+  if (fallbackToWindowLocation) {
+    setPreview && setPreview(true)
```

Comment:
```suggestion
    if(setPreview) setPreview(true)
```

Reply:
hmmm, good point. Do you recommend creating 2 functions, showPageLoading and hidePageLoading or were you thinking to keep the param (true/false) to set the preview state?